prow/plugins/lgtm/lgtm_test.go

```diff
@@ -50,6 +51,19 @@ type fakeRepoOwners struct {
 	reviewers map[string]sets.String
 }
 
+type fakePruner struct {
+	GithubClient  *fakegithub.FakeClient
+	IssueComments []github.IssueComment
+}
+
+func (fp *fakePruner) PruneComments(shouldPrune func(github.IssueComment) bool) {
+	for _, comment := range fp.IssueComments {
+		if shouldPrune(comment) {
+			fp.GithubClient.IssueCommentsDeleted = append(fp.GithubClient.IssueCommentsDeleted, comment.Body)
```

Comment:
Please use the `IssueCommentsDeleted` field to validate that the `LGTM added tree-sha:blahblah` comments are deleted.

Reply:
Added 2 tests for adding/removing this comment.